HW02_A_Graph-Fails.Rmd

```diff
@@ -1,87 +1,93 @@
+
+It is also best coding practice to only call the packages you use, so if you use a package but end up tossing the code you use for it, then make sure to remove loading it in the first place. For example, I could use `library("tidyverse")` but since this script will only be using ggplot2, I only load ggplot2.
+
+```{r load libraries}
+library("ggplot2")
+library("magrittr") #so I can do some piping
+```
+
+
+### Graph Fail 1
+
+What error is being thrown? How do you correct it? (hint, the error message tells you)
+
+```{r}
+data(mpg) #this is a dataset from the ggplot2 package
+
+mpg %>% 
+  ggplot(mapping = aes(x = "city", y = "hwy", color = "blue")) + 
+  geom_point()
+
+```
+
+### Graph Fail 2
+Why aren't the points blue? It is making me blue that the points in the graph aren't blue :`(
+```{r}
+
+ggplot(data = mpg) + 
+  geom_point(mapping = aes(x = displ, y = hwy ), color = "blue")
+
+```
+
+
+
+### Graph Fail 3
+Two mistakes in this graph. First, I wanted to make the the points slightly bolder, but changing the alpha to 2 does nothing. What does alpha do and what does setting it to 2 do? What could be done instead if I want the points slightly bigger? 
+
+Second, I wanted to move the legend on top of the graph since there aren't any points there, putting it at approximately the point/ordered pair (5, 40). How do you actually do this? Also, how do you remove the legend title ("class")? Finally, how would you remove the plot legend completely?
+```{r}
+# Alpha refers to the opacity of a geom
+mpg %>% 
+ggplot() + 
+  geom_point(mapping = aes(x = displ, y = hwy, color = class), size = 4) +
+  theme(legend.direction = "horizontal") + 
+  theme(legend.position = c(0.6,0.8))
+        
+#theme(legend.position = "none") is used to remove the legend completely
+#however, I am not sure how to put the legends at 5,40
+
+```
+
+### Graph Fail 4
+I wanted just one smoothing line. Just one line, to show the general relationship here. But that's not happening. Instead I'm getting 3 lines, why and fix it please?
+
+```{r}
+mpg %>% 
+ggplot(mapping = aes(x = displ, y = hwy, color = drv)) + 
+  geom_point() + 
+  geom_smooth(se = F, aes(color = NULL)) #se = F makes it so it won't show the error in the line of fit
+
+```
+
+### Graph Fail 5
+I got tired of the points, so I went to boxplots instead. However, I wanted the boxes to be all one color, but setting the color aesthetic just changed the outline? How can I make the box one color, not just the outline? 
+
+Also, the x-axis labels were overlaping, so I rotated them. But now they overlap the bottom of the graph. How can I fix this so axis labels aren't on the graph?
+
+
+```{r}
+ggplot(data = mpg, mapping = aes(x = manufacturer, y = cty)) + 
+  geom_boxplot(fill='#A4A4A4', color="black") + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

Comment:
I like the use of hjust. It's more elegant than using "vjust = 0.6"

Reply:
I agree! I used vjust=0.5, but when you use hjust, the spacing looks better